profiler/include/profile_convnd_bwd_data_impl.hpp

```diff
@@ -0,0 +1,439 @@
+        break;
+    default: break;
+    }
+}
+
+template <int NDimSpatial,
+          typename InDataType,
+          typename WeiDataType,
+          typename OutDataType,
+          typename InLayout,
+          typename WeiLayout,
+          typename OutLayout>
+void profile_convnd_bwd_data_impl(int do_verification,
+                                  int init_method,
+                                  bool do_log,
+                                  int nrepeat,
+                                  ck::index_t N,
+                                  ck::index_t K,
+                                  ck::index_t C,
+                                  std::vector<ck::index_t> input_spatial_lengths,
+                                  std::vector<ck::index_t> filter_spatial_lengths,
+                                  std::vector<ck::index_t> output_spatial_lengths,
+                                  std::vector<ck::index_t> conv_filter_strides,
+                                  std::vector<ck::index_t> conv_filter_dilations,
+                                  std::vector<ck::index_t> input_left_pads,
+                                  std::vector<ck::index_t> input_right_pads)
+{
+    using InElementOp  = ck::tensor_operation::element_wise::PassThrough;
+    using WeiElementOp = ck::tensor_operation::element_wise::PassThrough;
+    using OutElementOp = ck::tensor_operation::element_wise::PassThrough;
+
+    const auto in_element_op  = InElementOp{};
+    const auto wei_element_op = WeiElementOp{};
+    const auto out_element_op = OutElementOp{};
+
+    std::vector<std::size_t> input_dims{static_cast<std::size_t>(N), static_cast<std::size_t>(C)};
+    input_dims.insert(
+        std::end(input_dims), std::begin(input_spatial_lengths), std::end(input_spatial_lengths));
+
+    std::vector<std::size_t> filter_dims{static_cast<std::size_t>(K), static_cast<std::size_t>(C)};
+    filter_dims.insert(std::end(filter_dims),
+                       std::begin(filter_spatial_lengths),
+                       std::end(filter_spatial_lengths));
+
+    std::vector<std::size_t> output_dims{static_cast<std::size_t>(N), static_cast<std::size_t>(K)};
+    output_dims.insert(std::end(output_dims),
+                       std::begin(output_spatial_lengths),
+                       std::end(output_spatial_lengths));
+
+    Tensor<InDataType> in_n_c_hi_wi_host_result(
+        GetInputHostTensorDescriptor(input_dims, NDimSpatial));
+    Tensor<InDataType> in_n_c_hi_wi_device_result(
+        GetInputHostTensorDescriptor(input_dims, NDimSpatial));
+    Tensor<WeiDataType> wei_k_c_y_x(GetFiltersHostTensorDescriptor(filter_dims, NDimSpatial));
+    Tensor<OutDataType> out_n_k_ho_wo(GetOutputHostTensorDescriptor(output_dims, NDimSpatial));
+
+    std::cout << "in_n_c_hi_wi: " << in_n_c_hi_wi_host_result.mDesc << std::endl;
+    std::cout << "wei_k_c_y_x: " << wei_k_c_y_x.mDesc << std::endl;
+    std::cout << "out_n_k_ho_wo: " << out_n_k_ho_wo.mDesc << std::endl;
```

Comment:
All these tensor names are for 2D. Could be misleading. I don't have better suggestions, but others may do.